pagopa/io-app-design-system
# [IOAPPX-415] Remove the legacy UI + Add the new `useIONewTypeface` hook to manage new and legacy typography

## Short description
This PR completely removes the legacy UI by removing all the logic introduced to manage the new and old views simultaneously. It also adds the new `useIONewTypeface` hook to manage the legacy and new typography.

## List of changes proposed in this pull request
- Remove the legacy styles from all the components
- Replace `isExperimental` with `newTypefaceEnabled` in all the typographic styles

## How to test
1. Launch the example app
2. You should no longer need to activate the experimental DS
3. Test the components by enabling/disabling the new typeface toggle
4. Check that all the components are OK
